app/assets/javascripts/discourse/app/components/composer-body.js

```diff
@@ -146,9 +145,12 @@ export default Component.extend(KeyEnterEscape, {
       this.appEvents.trigger("composer:resize-started");
     });
 
-    if (iOSWithVisualViewport()) {
+    if (this._visualViewportResizing()) {
       this.viewportResize();
-      window.visualViewport.addEventListener("resize", this.viewportResize);
+      window.visualViewport.addEventListener(
+        "resize",
+        this.viewportResize.bind(this)
```

Comment:
This is incompatible with `removeEventListener` in `willDestroyElement`.

~~You'll have to save the bound function (`this._boundViewportResize = this.viewportResize.bind(this)`) and use it in both `addEventListener` and `removeEventListener`.~~

Better yet, we have `@bind` decorator (`import { bind } from "discourse-common/utils/decorators"`) to bind functions like `viewportResize`.

Reply:
Cool, I didn't know we had that decorator.

Reply:
I always forget we have it, and apparently I'm the one who [added](https://github.com/discourse/discourse/pull/10439) it! 😂